purchase_product_pack/models/purchase_order.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def copy(self, default=None):
+        purchase_copy = super().copy(default)
+        # we unlink pack lines that should not be copied
+        pack_copied_lines = purchase_copy.order_line.filtered(
+            lambda l: l.pack_parent_line_id.order_id == self
+        )
+        pack_copied_lines.unlink()
+        return purchase_copy
+
+    @api.onchange("order_line")
+    def check_pack_line_unlink(self):
+        """At least on embeded tree editable view odoo returns a recordset on
+        _origin.order_line only when lines are unlinked and this is exactly
+        what we need
+        """
```

Comment:
not a docstring explaining what the method does
```suggestion
    def check_pack_line_unlink(self):
        # At least on embeded tree editable view odoo returns a recordset on
        # _origin.order_line only when lines are unlinked and this is exactly
        # what we need
        """
```